src/sql/base/query/browser/untitledQueryEditorInput.ts

```diff
@@ -44,6 +48,28 @@ export class UntitledQueryEditorInput extends QueryEditorInput implements IUntit
 		return this.text.model.hasAssociatedFilePath;
 	}
 
+	override async save(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		// Create our own FileQueryEditorInput wrapper here so that the existing state (connection, results, etc) can be transferred from this input to the new file input.
```

Comment:
save and saveAs looks the same including the comments, you could have private method and both save and saveAs can call it.